### ISSUE TITLE
[Bug]: toPNG does not throw an error if it can't load an image (eg 404)

### ISSUE THREAD

**Issue by floklein** [clientIO/joint#2167]
### What happened?

Hi!
I have a paper with some images with invalid links that 404. When I call toPNG, nothing happens (except the 404 in the console).
toPNG only provides a callback that is not executed if it fails. I may not understand how it works, but couldn't it provide an error callback, or throw an error?
Thanks!

### Version

3.4.4

### What browsers are you seeing the problem on?

Chrome

### What operating system are you seeing the problem on?

Mac

**Comment by kumilingus** [comment]
I assume you use the `convertImagesToDataUris` option. The callback should be executed even if any of the conversions fail. We'll look into it.

**Comment by kumilingus** [comment]
Is this happening with the latest version?

**Comment by floklein** [comment]
The only "weird" thing I do is to wrap toPNG, to use as promise:
```js
const toPNGAndPatchAsync = () =>
  new Promise((resolve, reject) => {
    paper.toPNG(
      async preview => {
        try {
          // async api call
          resolve();
        } catch (err) {
          // error handling
          reject();
        }
      },
      { useComputedStyles: false, backgroundColor: 'transparent' },
    );
  });
```

**Comment by kumilingus** [comment]
Fixed in `v3.7` (about to be released).

```js
const toPNGAndPatchAsync = () =>
  new Promise((resolve, reject) => {
    paper.toPNG(
      async (preview, conversionError) => {
        if (conversionError) {
          // notify the user
        }
        try {
          // async api call
          resolve();
        } catch (err) {
          // error handling
          reject();
        }
      },
      { useComputedStyles: false, backgroundColor: 'transparent' },
    );
  });

```

**Comment by floklein** [comment]
Thanks a lot!

**Comment by kumilingus** [comment]
Fixed in `v3.7`.
